File: concrete/src/main/java/pcb/uwu/unit/derived/mechanics/NewtonUnit.java
package pcb.uwu.unit.derived.mechanics;

import static pcb.uwu.unit.base.KiloGramUnit.KILOGRAM;
import static pcb.uwu.unit.base.MeterUnit.METER;
import static pcb.uwu.unit.base.SecondUnit.SECOND;

public class NewtonUnit extends ForceUnit {

	public static final NewtonUnit NEWTON = new NewtonUnit();

	public NewtonUnit() {
		super(KILOGRAM, METER, SECOND);
	}

	@Override
	public String getSymbol() {
		return "N";
	}

	@Override
	public String getSingularName() {
		return "newton";
	}

	@Override
	public String getPluralName() {
		return "newtons";
	}
}

File: implementation/src/main/java/pcb/uwu/unit/derived/mechanics/NewtonUnit.kt
package pcb.uwu.unit.derived.mechanics

import pcb.uwu.unit.base.KiloGramUnit
import pcb.uwu.unit.base.MeterUnit
import pcb.uwu.unit.base.SecondUnit

object NewtonUnit
    : ForceUnit(massUnit = KiloGramUnit,
                lengthUnit = MeterUnit,
                timeUnit = SecondUnit)
{
    override val symbol = "N"
    override val singularName = "newton"
    override val pluralName = "newtons"
}